src/Model/Nosto/Entity/Helper/ProductHelper.php

```diff
@@ -86,7 +86,7 @@ private function getCommonCriteria(): Criteria
     public function loadExistingParentProducts(
         array $existentParentProductIds,
         SalesChannelContext $context,
-    ): SalesChannelRepositoryIterator {
+    ): RepositoryIterator {
```

Comment:
Can we also switch to this within `getProductsIterator`?

Reply:
I will transfer the logic to that method as well.